src/main/webapp/js/finscholar/homepagecontroller/homepagecontroller.js

```diff
@@ -16,54 +16,122 @@
 
 goog.module('finscholar.homepagecontroller');
 
-const JsactionActionFlow = goog.require('jsaction.ActionFlow');
-const JsactionDispatcher = goog.require('jsaction.Dispatcher');
-const JsactionEventContract = goog.require('jsaction.EventContract');
-const GoogDom = goog.require('goog.dom');
-const GoogSoy = goog.require('goog.soy');
+const googDom = goog.require('goog.dom');
+const googSoy = goog.require('goog.soy');
+const jsactionActionFlow = goog.require('jsaction.ActionFlow');
+const jsactionDispatcher = goog.require('jsaction.Dispatcher');
+const jsactionEventContract = goog.require('jsaction.EventContract');
 const {homepage} = goog.require('finscholar.homepagecontroller.templates');
+const {CollegeListView} = goog.require('finscholar.collegelistview');
+const {CollegePageView} = goog.require('finscholar.collegepageview');
+const {ErrorPageView} = goog.require('finscholar.errorpageview');
 const {PageController} = goog.require('pagecontroller');
+const {ScholarshipListView} = goog.require('finscholar.scholarshiplistview');
+const {ScholarshipPageView} = goog.require('finscholar.scholarshippageview');
 
 /**
  * Class for the home page controller.
  * @public
  */
 class HomePageController extends PageController {
-  constructor() {
+
+  /** 
+   * @param {!Element} container The HTML div where the main frame is rendered.
+   */
+  constructor(container) {
     super();
-    this.eventContract_ = new JsactionEventContract();
+    /** @private @const {!Element} */
+    this.container_ = container;
+
+    /** @private @const {!jsactionEventContract} */
+    this.eventContract_ = new jsactionEventContract();
+
+    /** @private @const {!jsactionDispatcher} */
+    this.dispatcher_ = new jsactionDispatcher();
+
+    /** @private @const {!function(jsaction.ActionFlow): undefined} */
+    this.bindedNavbarOnclickHandler_ = this.handleNavbarOnclickEvent_.bind(this);
+
+    this.initJsaction_();
+
+    /** @private {number} */
+    this.navbarPageIndex_ = 0;
+
+    /** @private @const {!Array<CollegeListView, ScholarshipListView, CollegePageView, ScholarshipPageView, ErrorPageView>} */
+    this.TEMPLATE_HANDLERS_ = [
+                               new CollegeListView(), 
+                               new ScholarshipListView(), 
+                               new CollegePageView(), 
+                               new ScholarshipPageView(), 
+                               new ErrorPageView()
+                               ];
```

Comment:
Be careful having all these views initialized at the same time, especially the ones that are stateful and hold data from the server. You might have too much loaded in the client at once which will slow down your app and maybe crash the browser. Typically, you would create and destroy them as needed.

Reply:
Yeah thanks I just changed everything to class instead of instances